### PR TITLE
[release/v2.24] Bump version to v2.24.1

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This bumps the version to v2.24.1, in preparation of December 2023 patch releases.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind documentation
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/assets/config/changelog.json
Comment (Waseem826) [inline modules/web/src/assets/config/changelog.json:1]: There were also couple of UI fixes (https://github.com/kubermatic/dashboard/pull/6418 & https://github.com/kubermatic/dashboard/pull/6423) which I don't see here. Or do we not need them to be mentioned?

Comment (embik) [reply modules/web/src/assets/config/changelog.json:1]: Sorry, I missed those. Will add them, thanks for noticing.

**Pull request merged**